ui/shared/AssetSelector.qml

```diff
@@ -0,0 +1,166 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.13
+import "../imports"
+
+Item {
+    id: root
+    property var assets
+    property var selectedAsset: {
+        "symbol": "snt", "name": "", "value": "", "fiatValue": "", "address": ""
+    }
+    width: 86
+    height: 24
+
+    Select {
+        id: select
+        model: root.assets
+        width: parent.width
+        bgColor: Style.current.transparent
+        bgColorHover: Style.current.grey
+        anchors.left: parent.left
+        anchors.right: parent.right
+        caretRightMargin: 7
+        select.radius: 6
+        select.height: root.height
+        menu.width: 343
+        selectedItemView: Item {
+            anchors.fill: parent
+            SVGImage {
+                id: iconImg
+                anchors.left: parent.left
+                anchors.leftMargin: 4
+                sourceSize.height: 24
+                sourceSize.width: 24
+                anchors.verticalCenter: parent.verticalCenter
+                fillMode: Image.PreserveAspectFit
+                source: "../app/img/tokens/" + selectedAsset.symbol.toUpperCase() + ".png"
+            }
+
+            StyledText {
+                id: selectedTextField
+                text: selectedAsset.symbol.toUpperCase()
+                anchors.left: iconImg.right
+                anchors.leftMargin: 4
+                anchors.verticalCenter: parent.verticalCenter
+                font.pixelSize: 15
+                height: 22
+                verticalAlignment: Text.AlignVCenter
```

Comment:
The text color needs to be theme dependent

Reply:
@PascalPrecht `StyledText` already has a default `color` that is `textColor`